File: src/test/java/com/postbox/PostBox.java
package com.postbox;

import java.util.Scanner;

public class PostBox {
    public static void main(String[] args) {

        float mLength;
        float mWidth;
        int mHeight;

        Scanner sc = new Scanner(System.in);
        Box3 box3  = new Box3(23f, 14f, 13);
        Box5 box5  = new Box5(39.5f, 27.5f, 23);

        System.out.println("Please enter object's length: ");
        mLength = sc.nextFloat();
        System.out.println("Please enter object's width: ");
        mWidth  = sc.nextFloat();
        System.out.println("Please enter object's height: ");
        mHeight = sc.nextInt();

        if (box3.validate(mLength,mWidth,mHeight)){
            System.out.println("Box3");
        }else if(box5.validate(mLength,mWidth,mHeight)){
            System.out.println("Box5");
        }
    }
}

File: src/test/kotlin/com/postbox/PostBox.kt
package com.postbox

import java.util.*

fun main() {

    var mLength:Float
    var mWidth :Float
    var mHeight:Int

    var sc = Scanner(System.`in`)
    println("Please enter object's length: ")
    mLength = sc.nextFloat()
    println("Please enter object's width: ")
    mWidth  = sc.nextFloat()
    println("Please enter object's height: ")
    mHeight = sc.nextInt()
    var box3 = Box3kt(23f,14f,13)
    var box5 = Box5kt(39.5f,27.5f,23)

    if (box3.validate(mLength,mWidth,mHeight)){
        println("Box3")
    }else if(box5.validate(mLength,mWidth,mHeight)){
        println("Box5")
    }


}
class Box5kt(inLength: Float,inWidth:Float,inHeight:Int):Box(inLength,inWidth,inHeight){}
class Box3kt(inLength: Float,inWidth:Float,inHeight:Int):Box(inLength,inWidth,inHeight){}

open class Box(val inLength: Float,val inWidth:Float,val inHeight:Int){
    open fun validate(length : Float, width : Float, height : Int)
            =(inLength>=length && inWidth >=width && inHeight >= height)
}